pkg/confidence/models.go

```diff
@@ -60,46 +58,27 @@ func NewGeneralResolutionError(msg string) ResolutionError {
 
 type APIConfig struct {
 	APIKey string
-	Region APIRegion
+	APIResolveUrl string
 }
 
 func NewAPIConfig(apiKey string) *APIConfig {
 	return &APIConfig{
 		APIKey: apiKey,
-		Region: APIRegionGlobal,
+		APIResolveUrl: "https://resolver.confidence.dev",
 	}
 }
 
-const (
-	APIRegionEU     = iota
-	APIRegionUS     = iota
-	APIRegionGlobal = iota
-)
-
-// Private types below
-
-const euAPIURL = "https://resolver.eu.confidence.dev/v1"
-const usAPIURL = "https://resolver.us.confidence.dev/v1"
-const globalAPIURL = "https://resolver.confidence.dev/v1"
-
-func (r APIRegion) apiURL() string {
-	if r == APIRegionEU {
-		return euAPIURL
-	} else if r == APIRegionUS {
-		return usAPIURL
-	} else if r == APIRegionGlobal {
-		return globalAPIURL
+func NewAPIConfigWithUrl(apiKey, apiResolveUrl string) *APIConfig {
```

Comment:
Calling `APIConfig` directly doesn't allow to define default values for `APIResolveUrl`. Calling our constructor fixes that, but GO doesn't support optional arguments so I ended up creating a second function for when the URL is configured by the users